### PR TITLE
[DTM] Asset location

### PR DESCRIPTION
**Pull request by Vagdevi1998**
Document asset location feature details and bulk import failure changes.

### CODE REVIEW

File: content/dtm/default-properties.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  - app_development
+weight: 60
+---
+
+### Characteristics of default properties {#characteristics-of-default-properties}
+
+Default properties cannot be deleted. All the fields are not accessible for default properties, and this accessibility may vary depending on the specific property.
+
+### Location {#location}
+
```
Comment (lra-sag) [inline content/dtm/default-properties.md:17]: Add a brief summary of use of this property
```suggestion
Location property enables you to assign location (latitude and longitude) to an asset.

You can set the default values of latitude and longitude by entering corresponding values or alternatively using the map view.
```

File: content/dtm/default-properties.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+To set the default values for latitude and longitude using map view, click **Choose on Map** at the end of the section. Click the full screen icon at the top right corner of the map to view it in full screen.
+
+The marker is positioned at D&uuml;sseldorf, Germany by default. Click or drag the marker to the preferred position to select the default value for latitude and longitude. The marker on the map is positioned at an approximated location when either latitude or longitude value is missing. To close the map view, click **Close** above the map.
```
Comment (lra-sag) [inline content/dtm/default-properties.md:20]:
```suggestion
The marker is positioned at D&uuml;sseldorf, Germany by default. Click or drag the marker to the preferred position to select the default value for latitude and longitude. To close the map view, click **Close** above the map.

Note that the marker on the map is positioned at an approximated location when either latitude or longitude value is missing.
```

**Review by lra-sag — CHANGES_REQUESTED** [review]

File: content/dtm/default-properties.md
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/default-properties.md:1]: Added it as a separate section in asset properties

File: content/dtm/asset-hierarchy-bundle/creating-assets-bulk-import.md [hunk 3]
```diff
@@ -171,6 +171,8 @@ To upload the CSV template follow the steps below:
 If the inputs are not provided in the correct format, the bulk import fails with validation errors, resulting in a **Failed asset imports** notification, showing the issue types, for example, an invalid template or an incorrect asset hierarchy.
 No assets are created.
 
+In case of bulk operation failure, the entire operation is rolled back and user will be navigated to main assets page to cleanup the entities created and try the operation again.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/asset-hierarchy-bundle/creating-assets-bulk-import.md]: Moved this to creating-assets-UI file

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md [hunk 4]
```diff
@@ -30,10 +30,15 @@ Root asset models are asset models which are at the top of hierarchy and don't h
 
 6. Under **Asset properties**, fill in the required information. Here you find all [asset properties](/dtm/asset-types/#asset-properties) assigned to the asset model that you create.
 
-    {{< c8y-admon-info>}}
- The asset properties are defined when [creating the asset model](/dtm/asset-types/#to-create-an-asset-model).
- When creating an asset you must provide the values for all asset properties.
-    {{< /c8y-admon-info>}}
+For an asset with a location property click **Choose on Map** in the section to set the values for latitude and longitude using map view. Click the full screen icon at the top right corner of the map to view it in full screen.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/asset-hierarchy-bundle/creating-assets-ui.md]: Modified

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md [hunk 5]
```diff
@@ -30,10 +30,15 @@ Root asset models are asset models which are at the top of hierarchy and don't h
...
+
+The marker is positioned at the default value set in the location property. If no defualt value is set,the marker is positioned at D&uuml;sseldorf, Germany. Click or drag the marker to the preferred position to select the value for latitude and longitude. The marker on the map is positioned at an approximated location when either latitude or longitude value is missing. To close the map view, click **close** above the map.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/asset-hierarchy-bundle/creating-assets-ui.md]: Modified

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 6]
```diff
@@ -17,6 +17,7 @@ helpContent:
 
 Asset properties are the parameters that define an [asset model](#asset-models). One or more asset properties are used to define an asset model. Asset properties are created in the **Asset properties** page.
 
+Default properties are available once you login into the tenant. See [Default properties](/dtm/default-properties/#characteristics-of-default-properties) to know the list of properties that are available.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: Modified

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 7]
```diff
@@ -297,6 +294,7 @@ Follow the steps below:
 Use the downloaded file to import the asset properties into a DTM application on another tenant.
 
 {{< c8y-admon-info>}}
+Default properties are not listed under Export asset properties. See [Default properties](/dtm/default-properties/#characteristics-of-default-properties) to know the list of properties that are available.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: Modified

File: content/dtm/default-properties.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+layout: bundle
+outputs:
+  - html
+  - json
+section:
+  - app_development
+weight: 60
+---
+
+### Characteristics of default properties {#characteristics-of-default-properties}
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/default-properties.md]: Modified

File: content/dtm/default-properties.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+Default properties cannot be deleted. All the fields are not accessible for default properties, and this accessibility may vary depending on the specific property.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/default-properties.md]: Modified

File: content/dtm/default-properties.md (hunk 1 above)
Comment (Vagdevi1998) [reply content/dtm/default-properties.md]: Added

File: content/dtm/default-properties.md (hunk 2 above)
Comment (Vagdevi1998) [reply content/dtm/default-properties.md]: Modified

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md [hunk 10]
```diff
@@ -30,10 +30,15 @@ Root asset models are asset models which are at the top of hierarchy and don't h
 
 6. Under **Asset properties**, fill in the required information. Here you find all [asset properties](/dtm/asset-types/#asset-properties) assigned to the asset model that you create.
 
-    {{< c8y-admon-info>}}
- The asset properties are defined when [creating the asset model](/dtm/asset-types/#to-create-an-asset-model).
- When creating an asset you must provide the values for all asset properties.
-    {{< /c8y-admon-info>}}
+For an asset with a location property, click **Choose on Map** in the section to set the values for latitude and longitude using map view. Click the full screen icon at the top right corner of the map to view it in full screen.
+
+The marker is positioned at the default value set in the location property. If no default value is set, the marker is positioned at D&uuml;sseldorf, Germany. Click or drag the marker to the preferred position to set the value for latitude and longitude. The marker on the map is positioned at an approximated location when either latitude or longitude value is missing. To close the map view, click **Close** above the map.
```
Comment (MWindel) [inline content/dtm/asset-hierarchy-bundle/creating-assets-ui.md:35]:
```suggestion
The marker is positioned at the default value set in the **Location** property. If no default value is set, the marker is positioned at Duesseldorf, Germany. Click or drag the marker to the preferred position to set the value for both latitude and longitude. The marker on the map is positioned at an approximated location when either the latitude or the longitude value is missing.
```

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md [hunk 11]
```diff
@@ -30,10 +30,15 @@ Root asset models are asset models which are at the top of hierarchy and don't h
 
 6. Under **Asset properties**, fill in the required information. Here you find all [asset properties](/dtm/asset-types/#asset-properties) assigned to the asset model that you create.
 
-    {{< c8y-admon-info>}}
- The asset properties are defined when [creating the asset model](/dtm/asset-types/#to-create-an-asset-model).
- When creating an asset you must provide the values for all asset properties.
-    {{< /c8y-admon-info>}}
+For an asset with a location property, click **Choose on Map** in the section to set the values for latitude and longitude using map view. Click the full screen icon at the top right corner of the map to view it in full screen.
```
Comment (MWindel) [inline content/dtm/asset-hierarchy-bundle/creating-assets-ui.md:33]:
```suggestion
For an asset with a **Location** property, click **Choose on Map** in the section to set the values for the latitude and longitude using map view. Click the full screen icon at the top right corner of the map to view it in full screen.
```

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md [hunk 12]
```diff
@@ -30,10 +30,15 @@ Root asset models are asset models which are at the top of hierarchy and don't h
...
+
+The marker is positioned at the default value set in the location property. If no default value is set, the marker is positioned at D&uuml;sseldorf, Germany. Click or drag the marker to the preferred position to set the value for latitude and longitude. The marker on the map is positioned at an approximated location when either latitude or longitude value is missing. To close the map view, click **Close** above the map.
+
+{{< c8y-admon-info>}}
+The values of latitude and longitude are automatically updated whenever a new location is selected on the map and vice-versa. The Altitude value is not represented on the map.<br>
```
Comment (MWindel) [inline content/dtm/asset-hierarchy-bundle/creating-assets-ui.md:38]:
```suggestion
The values of latitude and longitude are automatically updated whenever a new location is selected on the map and vice-versa. The altitude value is not represented on the map.<br>
```

File: content/dtm/asset-hierarchy-bundle/viewing-assets.md [hunk 13]
```diff
@@ -32,7 +34,8 @@ You see a warning message if one or more properties associated with the asset ar
 ### To modify an asset {#to-modify-an-asset}
 
 Fields are marked editable by an edit icon.
-To modify any of the asset details, click the edit icon, enter new details and save.
+To modify any of the asset details, click the edit icon, enter new details and save.<br>
+For an asset with a location, Click or drag the marker to the preferred position to select the value for latitude and longitude on the map. Click the full screen icon at the top right corner of the map to enter full screen mode. Use Escape key to exit from the full screen mode.
```
Comment (MWindel) [inline content/dtm/asset-hierarchy-bundle/viewing-assets.md:38]:
```suggestion
For an asset with a location, click or drag the marker to the preferred position to select the value for latitude and longitude on the map. Click the full screen icon at the top right corner of the map to enter full screen mode.```

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 14]
```diff
@@ -323,3 +319,24 @@ Review the errors that are reported, resolve them, and attempt the upload again.
...
+### Characteristics {#characteristics-of-default-properties}
+
+* Default properties are readily available when you open the application.
+* Default properties cannot be deleted. 
+* Only selected fields within a default property are editable and the editable fields vary depending on the specific property.
+* Default properties cannot be exported.
+
+### Location {#location}
+
+Location property enables you to assign location (latitude and longitude) to an asset. You can set the default values of latitude and longitude by entering corresponding values or alternatively using the map view.
```
Comment (MWindel) [inline content/dtm/asset-types-bundle/asset-properties.md:334]:
```suggestion
The **Location** property enables you to assign location (latitude and longitude) to an asset. You can set the default values of the latitude and longitude by entering the corresponding values or alternatively using the map view.
```

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 15]
```diff
@@ -323,3 +319,24 @@ Review the errors that are reported, resolve them, and attempt the upload again.
...
+
+To set the default values for latitude and longitude using map view, click **Choose on Map** at the end of the section. Click the full screen icon at the top right corner of the map to view it in full screen. The marker is positioned at D&uuml;sseldorf, Germany by default. Click or drag the marker to the preferred position to select the default value for latitude and longitude. To close the map view, click **Close** above the map.
```
Comment (MWindel) [inline content/dtm/asset-types-bundle/asset-properties.md:336]:
```suggestion
To set the default values for the latitude and longitude using the map view, click **Choose on Map** at the end of the section. Click the full screen icon at the top right corner of the map to view it in full screen. The marker is positioned at Duesseldorf, Germany by default. Click or drag the marker to the preferred position to select the default value for latitude and longitude.
```

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 16]
```diff
@@ -323,3 +319,24 @@ Review the errors that are reported, resolve them, and attempt the upload again.
...
+
+{{<c8y-admon-info>}}
+The valid range for Latitude is -90 to +90 and valid range for Longitude is -180 to +180. <br>
```
Comment (MWindel) [inline content/dtm/asset-types-bundle/asset-properties.md:339]:
```suggestion
The valid range for the latitude is -90 to +90 and valid range for the longitude is -180 to +180. <br>
```

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 17]
```diff
@@ -323,3 +319,24 @@ Review the errors that are reported, resolve them, and attempt the upload again.
...
+The marker on the map is positioned at an approximated location when either latitude or longitude value is missing.<br>
```
Comment (MWindel) [inline content/dtm/asset-types-bundle/asset-properties.md:340]:
```suggestion
The marker on the map is positioned at an approximated location when either the latitude or longitude value is missing.<br>
```

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 18]
```diff
@@ -323,3 +319,24 @@ Review the errors that are reported, resolve them, and attempt the upload again.
...
+The default values of latitude and longitude are automatically updated whenever a new location is selected on the map and vice-versa. The Altitude value is not represented on the map.
```
Comment (MWindel) [inline content/dtm/asset-types-bundle/asset-properties.md:341]:
```suggestion
The default values of both latitude and longitude are automatically updated whenever a new location is selected on the map and vice-versa. The Altitude value is not represented on the map.
```

Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: Modified

File: content/dtm/asset-types-bundle/asset-properties.md (hunk 17 above)
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: Modified

File: content/dtm/asset-types-bundle/asset-properties.md (hunk 16 above)
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: Modified

File: content/dtm/asset-types-bundle/asset-properties.md (hunk 15 above)
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: @MWindel is it not necessary to add Umlaut for Duesseldorf?

File: content/dtm/asset-types-bundle/asset-properties.md (hunk 14 above)
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: Modified

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 19]
```diff
@@ -310,10 +308,8 @@ To import asset properties into the DTM application, follow the steps below:
 2. Upload the JSON file in the drop file area of the resulting dialog. The JSON file is the file containing exported asset properties from a DTM application on another tenant.
 3. Click **Next** to preview the asset properties before importing them.
 4. Click **Import** to import the asset properties.
-4. Alternatively, Click **Back** to go to previous step. Hover over the file name and click the **Delete** icon to remove the current file and upload a different one.
-5. View newly imported asset properties in the **Asset properties** page.
-
-
+5. Alternatively, Click **Back** to go to previous step. Hover over the file name and click the **Delete** icon to remove the current file and upload a different one.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: Modified

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 20]
```diff
@@ -297,6 +294,7 @@ Follow the steps below:
 Use the downloaded file to import the asset properties into a DTM application on another tenant.
 
 {{< c8y-admon-info>}}
+Default properties are not listed under Export asset properties. See [Default properties](/dtm/default-properties/#characteristics-of-default-properties) to view the list of properties available.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md]: Formatted Export asset properties button to bold

File: content/dtm/asset-hierarchy-bundle/viewing-assets.md (hunk 13 above)
Comment (Vagdevi1998) [reply content/dtm/asset-hierarchy-bundle/viewing-assets.md]: Modified

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md [hunk 21]
```diff
@@ -119,6 +124,8 @@ On successful asset hierarchy creation, you see a pop-up notification in the top
 You can view the newly created asset hierarchy in the **Assets** page.
 Also see [viewing assets](/dtm/asset-hierarchy/#viewing-assets).
 
+In case of bulk operation failure, the entire operation is rolled back and user will be navigated to main assets page to cleanup the entities created and try the operation again.
```
[the remark this answers is not in the document]
Comment (Vagdevi1998) [reply content/dtm/asset-hierarchy-bundle/creating-assets-ui.md]: Modified

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md (hunk 12 above)
Comment (Vagdevi1998) [reply content/dtm/asset-hierarchy-bundle/creating-assets-ui.md]: Modified

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md (hunk 11 above)
Comment (Vagdevi1998) [reply content/dtm/asset-hierarchy-bundle/creating-assets-ui.md]: Modified

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md (hunk 10 above)
Comment (Vagdevi1998) [reply content/dtm/asset-hierarchy-bundle/creating-assets-ui.md]: @MWindel  is it necessary to add Umlaut for Duesseldorf?

File: content/dtm/asset-types-bundle/asset-properties.md [hunk 22]
```diff
@@ -323,3 +319,24 @@ Review the errors that are reported, resolve them, and attempt the upload again.
...
+* Default properties cannot be exported.
+
+### Location {#location}
+
+The **Location** property enables you to assign location (latitude and longitude) to an asset. You can set the default values of the latitude and longitude by entering the corresponding values or alternatively using the map view.
+
+To set the default values for the latitude and longitude using the map view, click **Choose on Map** at the end of the section. Click the full screen icon at the top right corner of the map to view it in full screen. The marker is positioned at Duesseldorf, Germany by default. Click or drag the marker to the preferred position to select the default value for latitude and longitude.
+
+{{<c8y-admon-info>}}
+The valid range for the latitude is -90 to +90 and valid range for the longitude is -180 to +180. <br>
```
Comment (MWindel) [inline content/dtm/asset-types-bundle/asset-properties.md:339]:
```suggestion
The valid range for the latitude is -90° to +90° and valid range for the longitude is -180° to +180°. <br>
```

File: content/dtm/asset-types-bundle/asset-models.md [hunk 23]
```diff
@@ -220,9 +221,9 @@ To import asset models, follow the steps below:
 1. On the **Asset models** page, click **Import asset models**.
 2. Upload the JSON file in the drop file area of the resulting dialog. The JSON file is the file containing exported asset models from a DTM application on another tenant.
 3. Click **Next** to preview the asset models. Initially, all the asset models are collapsed. Click the expand icon to the left of each asset model to view the next level of child asset models.
-3. Click **Import** to import the asset models.
-4. Alternatively, click **Back** to go to previous step. Hover over the file name and click the **Delete** icon to remove the current file and upload a different one.
-4. The newly imported asset models are listed on the **Asset models** page.
+4. Click **Import** to import the asset models.
+5. Alternatively, click **Back** to go to previous step. Hover over the file name and click the **Delete** icon to remove the current file and upload a different one.
```
Comment (MWindel) [inline content/dtm/asset-types-bundle/asset-models.md:225]: @Vagdevi1998 could you provide a screenshot of this step?

File: content/dtm/asset-hierarchy-bundle/viewing-assets.md [hunk 24]
```diff
@@ -32,7 +34,8 @@ You see a warning message if one or more properties associated with the asset ar
 ### To modify an asset {#to-modify-an-asset}
 
 Fields are marked editable by an edit icon.
-To modify any of the asset details, click the edit icon, enter new details and save.
+To modify any of the asset details, click the edit icon, enter new details and save.<br>
+For an asset with a location, click or drag the marker to the preferred position to select the value for latitude and longitude on the map. Click the full screen icon at the top right corner of the map to enter full screen mode. Use Escape key to exit from the full screen mode.
```
Comment (MWindel) [inline content/dtm/asset-hierarchy-bundle/viewing-assets.md:38]:
```suggestion
For an asset with a location, click or drag the marker to the preferred position to select the value for latitude and longitude on the map. Click the full screen icon at the top right corner of the map to enter full screen mode.
```

File: content/dtm/asset-hierarchy-bundle/creating-assets-ui.md (hunk 10 above)
Comment (MWindel) [reply content/dtm/asset-hierarchy-bundle/creating-assets-ui.md]: @Vagdevi1998 If you spell it "ue" is it fine.

File: content/dtm/asset-types-bundle/asset-properties.md (hunk 22 above)
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-properties.md:339]: Latitude and Longitude are fields which accepts both negative and position numbers within that range. 
@MWindel Should I still modify that

**Review by lra-sag — APPROVED** [review]

File: content/dtm/asset-types-bundle/asset-models.md (hunk 23 above)
Comment (Vagdevi1998) [reply content/dtm/asset-types-bundle/asset-models.md:225]: It is already there.

Comment (MWindel) [reply content/dtm/asset-types-bundle/asset-models.md:225]:
```suggestion
5. Alternatively, click **Back** to go to previous step. Hover over the file name and click the remove icon to remove the current file and upload a different one.
```

**Review by MWindel — APPROVED** [review]

**Pull request merged**